### PR TITLE
245 db limits and 292 concept DOI validation

### LINKED ISSUE #245: Limits on database columns
Most `VARCHAR` columns in the database have no size limit. We should impose limits on all columns in order to prevent uploading of arbitrary large amounts of data

### LINKED ISSUE #292: Software - edit information - concept DOI validation
On the software edit page we have place to input concept DOI. Some scrapers use this info to scrape citation information. The concept and versioned DOI's are usually provided by DataCite. We can validate if DOI is concept DOI and in case that it is version DOI we can extact the concept DOI using DataCite api.

The information about quering DataCite using GraphQL is provided in issue #289. This information can help build FE service to validate DOI and suggest concept DOI when version DOI is provided.

For the start we can add button "Validate DOI" below input. If not valid concept DOI we can use snackbar (warning type) and suggest concept DOI we find in version doi. Other design suggestions are welcome ;-)

image

### PR DESCRIPTION
**Pull request by dmijatovic**
# Define the limits for all VARCHAR columns in the database AND concept DOI validation button

Closes #245 
Closes #292 

Changes proposed in this pull request:
*  VARCHAR limits defined in all tables. Title, subtitle and description set to 200, 300 en 10.000 for both software and projects
*  TIMESTAMPTZ (time with timezone indication) applied
*  Forms using these columns show allowed max. length to the user
*  Find contributor, organisation or team member triggers after user types 2 characters
*  On edit software page, after DOI is provided, `validate DOI` button appears for user to confirm that it is valid concept DOI
*  If DOI is version DOI, like 10.5281/zenodo.6592693, and we can extract Concept DOI the button `UPDATE CONCEPT DOI` appears instead of Validate button.

How to test:
* `docker-compose down --volumes` remove old
* `docker-compose build`: build new
* `docker-compose up`: start
*  run data migrations script, it should not fail
*  create new software, try using values up to defined limits in all sections (info, contributors, organisations).
*  Add this DOI (version doi) and use validate DOI button, you should get warning message and the button update concept doi should appear
*  create new project, try using values up to defined limits in all sections (info, team, organisations).

## Example project input on "max"

image

## Example validate concept DOI

image

PR Checklist:

*   [x] Increase version numbers in `docker-compose.yml`
*   [x] Link to a GitHub issue
*   [ ] Update documentation
*   [ ] Tests

### CODE REVIEW

File: frontend/components/software/edit/information/ValidateConceptDoi.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    const info = await getSoftwareVersionInfoForDoi(doi)
+    if (info) {
+      if (info.versionOfCount === 0) {
+        showSuccessMessage(`Valid concept DOI: ${doi}`)
+      } else if (info.versionOfCount === 1) {
+        const conceptDoi = info.versionOf.nodes[0].doi
+        showErrorMessage(`This is version DOI. The concept DOI is ${conceptDoi}`)
+      }
+    } else {
+      showErrorMessage(`Failed to retereive version info for DOI: ${doi}. This is high likely not concept DOI.`)
```
Comment (cmeessen) [inline frontend/components/software/edit/information/ValidateConceptDoi.tsx:30]: I suggest to rephrase a little bit:

```suggestion
      showErrorMessage(`Failed to retrieve version info for DOI: ${doi}. It looks like this is not a Concept DOI.`)
```

File: frontend/components/software/edit/information/ValidateConceptDoi.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  async function validateDoi() {
+    setLoading(true)
+    const info = await getSoftwareVersionInfoForDoi(doi)
+    if (info) {
+      if (info.versionOfCount === 0) {
+        showSuccessMessage(`Valid concept DOI: ${doi}`)
+      } else if (info.versionOfCount === 1) {
+        const conceptDoi = info.versionOf.nodes[0].doi
+        showErrorMessage(`This is version DOI. The concept DOI is ${conceptDoi}`)
```
Comment (cmeessen) [inline frontend/components/software/edit/information/ValidateConceptDoi.tsx:27]: Minor addition:

```suggestion
        showErrorMessage(`This is a version DOI. The Concept DOI is ${conceptDoi}`)
```

File: frontend/components/software/edit/information/ValidateConceptDoi.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+export default function ValidateConceptDoi({doi}: { doi: string }) {
+  const {showErrorMessage,showSuccessMessage, showInfoMessage} = useSnackbar()
+  const [loading,setLoading]=useState(false)
+
+  async function validateDoi() {
+    setLoading(true)
+    const info = await getSoftwareVersionInfoForDoi(doi)
+    if (info) {
+      if (info.versionOfCount === 0) {
+        showSuccessMessage(`Valid concept DOI: ${doi}`)
```
Comment (cmeessen) [inline frontend/components/software/edit/information/ValidateConceptDoi.tsx:24]: Minor rephrase suggestion:

```suggestion
        showSuccessMessage(`The DOI ${doi} is a valid Concept DOI.`)
```

File: frontend/components/software/edit/editSoftwareConfig.ts [hunk 4]
```diff
@@ -210,7 +226,7 @@ export const organisationInformation = {
     title: 'Add organisation',
     subtitle: 'We search organisation name in RSD and ROR databases',
     label: 'Find or add organisation',
-    help: 'At least fist 3 letters of organisation name',
+    help: 'At least fist 2 letters of organisation name',
```
Comment (cmeessen) [inline frontend/components/software/edit/editSoftwareConfig.ts:229]: Typo :)

```suggestion
    help: 'At least the first 2 letters of the organisation name',
```

File: frontend/components/software/edit/editSoftwareConfig.ts [hunk 5]
```diff
@@ -210,7 +226,7 @@ export const organisationInformation = {
     title: 'Add organisation',
     subtitle: 'We search organisation name in RSD and ROR databases',
```
Comment (cmeessen) [inline frontend/components/software/edit/editSoftwareConfig.ts:227]: While I am at it, here is one more suggestion :)

```suggestion
    subtitle: 'We search for organisation names in the RSD and the ROR databases',
```

File: frontend/components/software/edit/editSoftwareConfig.ts [hunk 6]
```diff
@@ -127,7 +143,7 @@ export const contributorInformation = {
     title: 'Add contributor',
     subtitle: 'We search by name in RSD and ORCID databases',
```
Comment (cmeessen) [inline frontend/components/software/edit/editSoftwareConfig.ts:144]: Here too:

```suggestion
    subtitle: 'We search by name in the RSD and the ORCID databases',
```

File: frontend/components/software/edit/editSoftwareConfig.ts [hunk 7]
```diff
@@ -112,7 +123,12 @@ export const softwareInformation = {
   licenses: {
     title: 'Licenses',
     subtitle: 'What licenses do apply to your software? You can also import licenses using concept DOI.',
-    help: 'Select license'
+    label: 'Find or add license',
```
Comment (cmeessen) [inline frontend/components/software/edit/editSoftwareConfig.ts:126]:
```suggestion
    label: 'Find or add a license',
```

**Review by cmeessen — COMMENTED** [review]
Discovering the Concept DOI from the DOI is pretty cool :+1:. I added plenty of smaller suggestions for rephrasing.

Suggestion: If the user provides a non-Concept DOI, we could provide a button to automatically replace the DOI.

@dmijatovic you added yourself as `SPDX-FileCopyrightText: 2022 Dusan Mijatovic` to each header and now you appear as a duplicate with `PDX-FileCopyrightText: 2022 Dusan Mijatovic (dv4all)`

**Review by ewan-escience — APPROVED** [review]

**Review by cmeessen — APPROVED** [review]
Perfect! Thank you for considering my suggestions :)

**Pull request merged**